crates/net/downloaders/src/headers/reverse_headers.rs

```diff
@@ -591,15 +652,25 @@ where
 
                     // Update the sync target hash
                     self.sync_target = match self.sync_target.take() {
-                        Some(mut sync_target) => {
-                            sync_target.hash = target;
-                            Some(sync_target)
-                        }
+                        Some(sync_target) => Some(sync_target.with_hash(target)),
                         None => Some(SyncTargetBlock::from_hash(target)),
                     };
                     self.on_block_number_update(parent_block_number, parent_block_number);
                 }
             }
+            SyncTarget::TipNum(num) => {
+                let current_tip_num = self.sync_target.as_ref().and_then(|t| t.number());
+                if Some(num) != current_tip_num {
+                    trace!(target: "downloaders::headers", %num, "Updating sync target based on num");
+                    self.metrics.out_of_order_requests.increment(1);
+                    // just update the sync target
+                    self.sync_target = Some(SyncTargetBlock::from_number(num));
+                    // self.on_block_number_update(num, num);
```

Comment:
does this need to get re-enabled or should it be removed?

Reply:
ah, should be removed